### PR TITLE
[feature] added find method

### PR DESCRIPTION
**Pull request by artemyankovets**
added find method

### CODE REVIEW

File: src/main/java/com/breskul/bibernate/persistence/JdbcDao.java [hunk 1]
```diff
@@ -3,27 +3,32 @@
...
 
-import static com.breskul.bibernate.persistence.util.DaoUtils.getClassTableName;
-import static com.breskul.bibernate.persistence.util.DaoUtils.getFieldValue;
-import static com.breskul.bibernate.persistence.util.DaoUtils.isValidEntity;
+import static com.breskul.bibernate.persistence.util.DaoUtils.*;
 
 public class JdbcDao {
+
     private static final Logger logger = LoggerFactory.getLogger(JdbcDao.class);
+    private static final String CHECK_YOUR_SQL_QUERY = "check your sql query";
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/persistence/JdbcDao.java]: How do you think about replace this error inside exception class?

File: src/main/java/com/breskul/bibernate/persistence/JdbcDao.java [hunk 2]
```diff
@@ -121,4 +164,89 @@ private void setPreparedValues(List<Object> values, PreparedStatement preparedSt
...
+        performReturningStatement(
+                ps -> {
+                    preparedStatementConsumer.accept(ps);
+                    return null;
+                },
+                cause
+        );
+    }
+
+    private <T, E extends SQLException> T performReturningStatement(
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/persistence/JdbcDao.java]: I think we should avoid this construction. In the next PR we will have only one connection for this class

File: src/main/java/com/breskul/bibernate/persistence/JdbcDao.java (hunk 1 above)
Comment (artemyankovets) [reply src/main/java/com/breskul/bibernate/persistence/JdbcDao.java]: done

**Review by Mikhail-Mikhailenko — APPROVED** [review]

File: src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java [hunk 3]
```diff
@@ -2,55 +2,153 @@
...
+        assertEquals(node.getPerson().getId(), selectedNote.getPerson().getId());
+
+        Assertions.assertDoesNotThrow(() -> entityManager.remove(selectedNote));
+        Assertions.assertThrows(JdbcDaoException.class, () -> entityManager.find(NoteComplex.class, node.getId()));
+
+        Assertions.assertDoesNotThrow(() -> entityManager.remove(selectedPerson));
+        Assertions.assertThrows(JdbcDaoException.class, () -> entityManager.find(Person.class, person.getId()));
+
+        entityTransaction.commit();
+    }
```
Comment (artemyankovets) [inline src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java:142]: Add one more test for Note when Person is null

**Review by yuriyfomin17 — CHANGES_REQUESTED** [review]
@artemyankovets third test does not pass 
Снимок экрана 2023-03-31 в 15 30 35

**Comment by artemyankovets** [comment]
> @artemyankovets third test does not pass Снимок экрана 2023-03-31 в 15 30 35

pls pull the updated version

**Review by yuriyfomin17 — APPROVED** [review]
